backend/uclapi/roombookings/views.py

```diff
@@ -163,3 +164,66 @@ def get_equipment(request, *args, **kwargs):
         "ok": True,
         "equipment": _serialize_equipment(equipment)
     }, rate_limiting_data=kwargs)
+
+
+@api_view(['GET'])
+@uclapi_protected_endpoint()
+def free_rooms(request, *args, **kwargs):
+    request_params = {}
+    request_params['startdatetime__gte'] = request.GET.get('start_datetime')
+    request_params['finishdatetime__lte'] = request.GET.get('end_datetime')
+
+    if (
+        not request_params['startdatetime__gte'] or
+        not request_params['finishdatetime__lte']
+    ):
+        return PrettyJsonResponse({
+            "ok": False,
+            "error": "start_datetime or end_datetime not provided"
+        })
+
+    is_parsed = True
+
+    start, end, is_parsed = _parse_datetime(
+        request_params['startdatetime__gte'],
+        request_params['finishdatetime__lte'],
+        None
+    )
+
+    if not is_parsed:
+        return PrettyJsonResponse({
+            "ok": False,
+            "error": "date/time isn't formatted as suggested in the docs"
+        })
+
+    # Rounding down start date to start of day
+    request_params["startdatetime__gte"] = _round_date(start)
+
+    # Rounding up end date to start of next day
+    request_params["finishdatetime__lte"] = _round_date(end, up=True)
+
+    # Pagination Logic
+    # maxing out results_per_page to get all the bookings in one page
+    results_per_page = 100000
+    request_params = dict((k, v) for k, v in request_params.items() if v)
```

Comment:
You can use a dictionary comprehension directly - `{k:v for k,v in request_params.items() if v}`